Dockerfile

```diff
@@ -1,73 +1,30 @@
-FROM alpine:3.12
-ENV AKTUALIZR_SRCREV 2020.10+fio
-WORKDIR /root/
-
-RUN apk add --no-cache cmake git g++ make curl-dev libarchive-dev libsodium-dev dpkg-dev doxygen graphviz sqlite-dev glib-dev autoconf automake libtool python3 boost-dev ninja \
-	&& git clone https://github.com/vlm/asn1c \
-	&& cd asn1c \
-	&& autoreconf -iv \
-	&& ./configure \
-	&& make -j`getconf _NPROCESSORS_ONLN` install \
-	&& cd ../ \
-	&& git clone https://github.com/foundriesio/aktualizr.git \
-	&& cd aktualizr \
-	&& git checkout $AKTUALIZR_SRCREV \
-	&& git submodule update --init --recursive \
-	&& mkdir build-git \
-	&& cd build-git \
-	&& sed -i '/GLOB_PERIOD/a #define GLOB_TILDE 4096' /usr/include/glob.h \
-	&& cmake -GNinja -DWARNING_AS_ERROR=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_SOTA_TOOLS=ON .. \
-	&& ninja src/sota_tools/install
-
-# download and extract OE utilities
-RUN cd /root/ \
-  && mkdir oe \
-  && wget https://github.com/openembedded/openembedded-core/archive/2020-04.1-dunfell.tar.gz -O oe.tar.gz \
-  && tar -xf oe.tar.gz --strip-components=1 -C oe \
-  && rm oe.tar.gz
-
-RUN apk add patch
-COPY 0001-wic-Adjust-cmd-line-format-to-debugfs-1.45.6.patch /root/oe
-RUN cd /root/oe \
-  && patch scripts/lib/wic/engine.py 0001-wic-Adjust-cmd-line-format-to-debugfs-1.45.6.patch
-
-RUN apk add python3 py3-pip libffi-dev make openssl-dev gcc libc-dev python3-dev rust cargo
-RUN pip3 install docker-compose==1.27 expandvars==0.6.5 awscli==1.20.64
-
-
 FROM golang:alpine
 RUN apk add gcc git glib-dev make musl-dev
 RUN git clone https://github.com/foundriesio/ostreeuploader.git /ostreeuploader && \
-	cd /ostreeuploader && git checkout 54daa6777667ab9263c65a379453b65d9a2eb19a
+	cd /ostreeuploader && git checkout 0372bd4386a28ed02a1cd9f643c9e58700db4367
 RUN cd /ostreeuploader && make
 
-FROM docker:20.10.2-dind
+FROM docker:20.10.12-dind
 WORKDIR /root/
 
-RUN apk add --no-cache bash glib libarchive libcurl libsodium nss openjdk8-jre-base ostree python3 boost-program_options boost-log boost-filesystem boost-log_setup parted sgdisk git lsblk skopeo
+ENV GARAGE_SIGN_ARCH="cli-0.7.2-48-gf606131.tgz"
+ENV GARAGE_SIGN_ARCH_HASH="f20c9f3e08fff277a78786025105298322c54874ca66a753e7ad0b2ffb239502"
+RUN wget https://storage.googleapis.com/public-shared-artifacts-fio/mirrors/ota-tuf-cli-releases/${GARAGE_SIGN_ARCH} \
+	&& echo "${GARAGE_SIGN_ARCH_HASH}  ${GARAGE_SIGN_ARCH}" | sha256sum -c \
+	&& tar -xzf ${GARAGE_SIGN_ARCH} --strip-components=1 -C /usr
+
+RUN apk add --no-cache bash glib libarchive libcurl libsodium nss openjdk8-jre-base python3 py3-pip boost-program_options boost-log boost-filesystem boost-log_setup parted sgdisk git lsblk skopeo ostree docker-compose \
+	&& pip3 install expandvars==0.6.5 awscli==1.20.64
```

Comment:
we could just pip install compose here in the final stage and not do it multi-stage.

Reply:
Yes, we can do it now since aktualizr compilation is removed. If we really need a version lower than the version installed by `docker:20.10.12-dind` via `apk add` (1.29.2) then we can switch to it. Until then doing it doesn't make much sense since it leads to docker-compose dependencies compilation which in turn requires installation of additional packages (e.g. gcc).